notebook/autobuild_function_calling.ipynb

```diff
@@ -0,0 +1,405 @@
+    "            )\n",
+    "\n",
+    "    return papers\n",
+    "\n",
+    "\n",
+    "list_of_functions = [\n",
+    "    {\n",
+    "        \"name\": \"get_arxiv_papers\",\n",
+    "        \"description\": \"Get arxiv papers published in last 24 hours\",\n",
+    "        \"function\": get_arxiv_papers,\n",
+    "    }\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 5: build group chat agents\n",
+    "\n",
+    "Use `build()` to let build manager (the specified `builder_model`) complete the group chat agents generation. Specify `list_of_functions` to be used by the Agents"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\u001b[32m==> Generating agents...\u001b[0m\n",
+      "['ArxivAPI_Expert', 'WebScraping_Expert', 'DataMining_Expert', 'Python_Expert', 'ResearchAnalysis_Expert'] are generated.\n",
+      "\u001b[32m==> Generating system message...\u001b[0m\n",
+      "Preparing system message for ArxivAPI_Expert\n",
+      "Preparing system message for WebScraping_Expert\n",
+      "Preparing system message for DataMining_Expert\n",
+      "Preparing system message for Python_Expert\n",
+      "Preparing system message for ResearchAnalysis_Expert\n",
+      "\u001b[32m==> Generating description...\u001b[0m\n",
+      "Preparing description for ArxivAPI_Expert\n",
+      "Preparing description for WebScraping_Expert\n",
+      "Preparing description for DataMining_Expert\n",
+      "Preparing description for Python_Expert\n",
+      "Preparing description for ResearchAnalysis_Expert\n",
+      "\u001b[32m==> Creating agents...\u001b[0m\n",
+      "Creating agent ArxivAPI_Expert...\n",
+      "Creating agent WebScraping_Expert...\n",
+      "Creating agent DataMining_Expert...\n",
+      "Creating agent Python_Expert...\n",
+      "Creating agent ResearchAnalysis_Expert...\n",
+      "Adding user console proxy...\n",
+      "Function get_arxiv_papers is registered to agent ArxivAPI_Expert.\n"
+     ]
+    }
+   ],
+   "source": [
+    "agent_list, agent_configs = builder.build(building_task, llm_config, list_of_functions)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here you can see that Function `exec_python` has been associated with `ArxivAPI_Expert` Agent."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 6: execute task\n",
+    "\n",
+    "Let agents generated in `build()` to complete the task collaboratively in a group chat."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\u001b[33mArxivAPI_Expert\u001b[0m (to chat_manager):\n",
+      "\n",
+      "Get arxiv papers published in last 24 hours\n",
```

Comment:
The task should reflect the collaboration between agents, such as "analysis papers in the field of computer science in the last 24 hours."